projects/packages/sync/src/class-sender.php

```diff
@@ -321,7 +321,35 @@ private function continue_full_sync_enqueue() {
 	 * @return boolean|WP_Error True if this sync sending was successful, error object otherwise.
 	 */
 	public function do_sync() {
-		return $this->do_sync_and_set_delays( $this->sync_queue );
+		if ( ! Settings::is_dedicated_sync_enabled() ) {
+			$result = $this->do_sync_and_set_delays( $this->sync_queue );
+		} else {
+			$result = Dedicated_Sender::spawn_sync( $this->sync_queue );
+		}
+
+		return $result;
+	}
+
+	/**
+	 * Trigger incremental sync and early exit on Dedicated Sync request.
+	 *
+	 * @access public
+	 */
+	public function do_dedicated_sync_and_exit() {
+		if ( ! Settings::is_dedicated_sync_enabled() ) {
+			return new WP_Error( 'dedicated_sync_disabled', 'Dedicated Sync flow is disabled.' );
+		}
+
+		if ( ! Dedicated_Sender::is_dedicated_sync_request() ) {
+			return new WP_Error( 'non_dedicated_sync_request', 'Not a Dedicated Sync request.' );
+		}
+
+		$result = $this->do_sync_and_set_delays( $this->sync_queue );
+		// If no errors occurred, re-spawn a dedicated Sync request.
+		if ( true === $result ) {
```

Comment:
Should there be a check for queue size before we trigger the next dedicated sync request?

Reply:
Ahh, nice it is so a check here would be redundent.

> However I'm still considering if it makes sense to attempt to spawn sync even if do_sync_and_set_delays returned an error given the existing checks in spawn_sync.

I think we should back off on error. The next process that comes along will trigger the send so lets not keep triggering a send that is going to fail fail fail and end up in a loop.